sdk/storage/azure-storage-blob/azure/storage/blob/aio/_container_client_async.py

```diff
@@ -136,6 +147,107 @@ def _build_generated_client(self):
         client._config.version = self._api_version # pylint: disable=protected-access
         return client
 
+    def _format_url(self, hostname):
+        return _format_url(
+            container_name=self.container_name,
+            hostname=hostname,
+            scheme=self.scheme,
+            query_str=self._query_str
+        )
+
+    @classmethod
+    def from_container_url(
+            cls, container_url: str,
+            credential: Optional[Union[str, Dict[str, str], "AzureNamedKeyCredential", "AzureSasCredential", "AsyncTokenCredential"]] = None,  # pylint: disable=line-too-long
+            **kwargs: Any
+        ) -> Self:
```

Comment:
Note sure if I missed this in other places but lines 160-163 have one extra indent. Also 208-212 below.

Reply:
Although indents seem consistently "wrong" in this file (as opposed to the style we chose for Queue when refactoring there) so probably can catch all this in typing pass or just leave it. Though I would love if we picked a consistent indent scheme for function defs.